mypy/semanal.py

```diff
@@ -1052,12 +1083,13 @@ def handle_missing_overload_implementation(self, defn: OverloadedFuncDef) -> Non
         """Generate error about missing overload implementation (only if needed)."""
         if not self.is_stub_file:
             if self.type and self.type.is_protocol and not self.is_func_scope():
-                # An overloaded protocol method doesn't need an implementation.
+                # An overloaded protocol method doesn't need an implementation,
+                # but if it doesn't have one, then it is considered implicitly abstract.
```

Comment:
I think an overload without implementation at all should be explicitly abstract, so you should use `IS_ABSTRACT` below.